File: dodam-core/src/main/java/b1nd/dodamcore/bus/domain/exception/BusExceptionCode.java
package b1nd.dodamcore.bus.domain.exception;

import b1nd.dodamcore.common.exception.ExceptionCode;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;

@AllArgsConstructor
public enum BusExceptionCode implements ExceptionCode {

    BUS_PERIOD_EXPIRED(HttpStatus.GONE,"기간 만료된 버스"),
    BUS_FULL_OF_SEAT(HttpStatus.BAD_REQUEST,"버스 좌석 만료"),
    BUS_ALREADY_APPLIED(HttpStatus.CONFLICT,"이미 신청한 버스"),
    BUS_MEMBER_NOT_FOUND(HttpStatus.NOT_FOUND, "없는 버스 멤버"),
    BUS_NOT_FOUND(HttpStatus.NOT_FOUND, "없는 버스");

    private final HttpStatus status;
    private final String message;

    @Override
    public HttpStatus getHttpStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }
}

File: services/service-bus/src/main/kotlin/com/b1nd/dodamdodam/bus/domain/bus/exception/BusExceptionCode.kt
package com.b1nd.dodamdodam.bus.domain.bus.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class BusExceptionCode(
    override val status: HttpStatus,
    override val message: String
): ExceptionCode {
    BUS_NOT_FOUND(HttpStatus.NOT_FOUND, "버스를 찾을 수 없어요."),
    BUS_APPLICANT_NOT_FOUND(HttpStatus.NOT_FOUND, "버스 신청자가 아니에요."),
    BUS_ALREADY_APPLIED_POSITION(HttpStatus.BAD_REQUEST, "이미 신청되어있는 자리예요."),
    USER_NOT_FOUND(HttpStatus.NOT_FOUND, "사용자를 찾을 수 없어요."),
    USER_SERVICE_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "사용자 서비스에 오류가 발생했어요."),
}
